epoch1/cuda/ee_mumu/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -421,30 +436,51 @@ int main(int argc, char **argv)
-        const int ieppM = ievt%neppM; // #event in the current eventpage in this iteration
-        std::cout << "Momenta:" << std::endl;
+      auto getMomentum = [&](std::size_t evtNo, int particle, int component) {
+        assert(component < mgOnGpu::np4);
+        assert(particle  < mgOnGpu::npar);
+        const auto page  = evtNo / neppM; // #eventpage in this iteration
+        const auto ieppM = evtNo % neppM; // #event in the current eventpage in this iteration
+        return hstMomenta[page * npar*np4*neppM + particle * neppM*np4 + component * neppM + ieppM];
+      };
+      auto dumpParticles = [&](std::ostream& stream, std::size_t evtNo, unsigned precision) {
+        const auto width = precision + 8;
         for (int ipar = 0; ipar < npar; ipar++)
         {
           // NB: 'setw' affects only the next field (of any type)
-          std::cout << std::scientific // fixed format: affects all floats (default precision: 6)
-                    << std::setw(4) << ipar + 1
-                    << std::setw(14) << hstMomenta[ipagM*npar*np4*neppM + ipar*neppM*np4 + 0*neppM + ieppM] // AOSOA[ipagM][ipar][0][ieppM]
-                    << std::setw(14) << hstMomenta[ipagM*npar*np4*neppM + ipar*neppM*np4 + 1*neppM + ieppM] // AOSOA[ipagM][ipar][1][ieppM]
-                    << std::setw(14) << hstMomenta[ipagM*npar*np4*neppM + ipar*neppM*np4 + 2*neppM + ieppM] // AOSOA[ipagM][ipar][2][ieppM]
-                    << std::setw(14) << hstMomenta[ipagM*npar*np4*neppM + ipar*neppM*np4 + 3*neppM + ieppM] // AOSOA[ipagM][ipar][3][ieppM]
-                    << std::endl
-                    << std::defaultfloat; // default format: affects all floats
+          stream << std::scientific // fixed format: affects all floats (default precision: 6)
+                 << std::setprecision(precision)
+                 << std::setw(4) << ipar + 1
+                 << std::setw(width) << getMomentum(ievt, ipar, 0)
+                 << std::setw(width) << getMomentum(ievt, ipar, 1)
+                 << std::setw(width) << getMomentum(ievt, ipar, 2)
+                 << std::setw(width) << getMomentum(ievt, ipar, 3)
+                 << std::endl
+                 << std::defaultfloat; // default format: affects all floats
         }
+      };
+
+      if (verbose)
+      {
+        // Display momenta
+        std::cout << "Momenta:" << std::endl;
+        dumpParticles(std::cout, ievt, 6);
         std::cout << std::string(80, '-') << std::endl;
+
         // Display matrix elements
         std::cout << " Matrix element = "
                   << hstMEs[ievt] << " GeV^" << meGeVexponent << std::endl; // FIXME: assume process.nprocesses == 1
         std::cout << std::string(80, '-') << std::endl;
       }
+
+      if (dumpFile.is_open())
+      {
+        dumpFile << "Event " << std::setw(8) << ievt << "  "
+                 << "Batch " << std::setw(4) << iiter << "\n";
+        dumpParticles(dumpFile, ievt, 15);
```

Comment:
why 6 to screen and 15 to file? is the actual precision 15 while 6 is a truncation?